packages/core/src/data/sources/lazy/axisMeasureSource.js

```diff
@@ -0,0 +1,154 @@
+     */
+    calculateCoord(alignMeasure, scale, measureDomainSize) {
+        let coordLeft, coordCenter, coordRight;
+        switch (alignMeasure) {
+            case "left":
+            case "bottom": // Fall-through case for 'bottom' along with 'left'
+                coordLeft = scale.domain()[0];
+                coordRight = coordLeft + measureDomainSize;
+                coordCenter = Math.floor((coordLeft + coordRight) / 2);
+                break;
+            case "center":
+                // @ts-ignore
+                coordCenter = scale.invert(0.5);
+                coordLeft = coordCenter - Math.floor(measureDomainSize / 2);
+                coordRight = coordCenter + Math.floor(measureDomainSize / 2);
+                break;
+            case "right":
+            case "top": // Fall-through case for 'top' along with 'right'
+                coordRight = scale.domain()[1];
+                coordLeft = coordRight - measureDomainSize;
+                coordCenter = Math.floor((coordLeft + coordRight) / 2);
+                break;
+            default:
+                console.error("Invalid alignMeasure value");
+                return null; // Return null or handle as necessary for invalid inputs
+        }
+        return [coordLeft, coordCenter, coordRight];
+    }
+
+    onDomainChanged() {
+        /** @type {ScaleWithProps} scale */
+        const scale = this.scaleResolution.scale;
+        const axisLength = this.getAxisLength();
+        const minMeasureSize = this.params.minMeasureSize;
+        const alignMeasure = this.params.alignMeasure;
+        const hideMeasureThreshold = this.params.hideMeasureThreshold;
+        // Find the size (in domain units) of a measure
+        // that shows minMeasureSize pixels wide
+        const minValueSize = Math.floor(
+            // @ts-ignore
+            scale.invert(minMeasureSize / axisLength) - scale.invert(0)
+        );
+        // Find the first measure size (in domain units) that produces a range
+        // bigger than the minimum requested size measure in pixels
+        const measureDomainSize = this.measureValues.find(
+            (value) => value > minValueSize
+        );
+        const measureLabel = this.measureLabeledValues[measureDomainSize];
+        // Position the measure according to the alignMeasure parameter
+        const [startPos, centerPos, endPos] = this.calculateCoord(
+            alignMeasure,
+            scale,
+            measureDomainSize
+        );
+        const measurePixelsSize = Math.floor(
+            (scale(endPos) - scale(startPos)) * axisLength
+        );
+        if (measureDomainSize <= hideMeasureThreshold) {
+            this.publishData([]); // hide the measure by publishing an empty array
+        } else {
+            if (startPos != this.startPos || endPos != this.endPos) {
+                this.publishData([
+                    [
+                        {
+                            startPos: startPos,
```

Comment:
When the property and variable names are the same, you can make it more succinct:

```js
{
  startPos,
  endPos,
  centerPos,
  ...
}
```